pkg/controller/stack/stack_controller.go

```diff
@@ -0,0 +1,981 @@
+			curVersion, err = semver.ParseTolerant(stack.stack.Version)
+
+			if err != nil {
+				log.Info(fmt.Sprintf("findMaxVersionStack: Invalid semver " + stack.stack.Version))
+			}
+		}
+
+		if err == nil {
+			if curVersion.Compare(maxVersion) > 0 {
+				maxStack = &stack
+				maxVersion = curVersion
+			}
+		}
+	}
+
+	// It's possible we didn't find a valid semver, in which case we'd return nil.
+	return maxStack
+}
+
+func (r *ReconcileStack) ensureStackHasOwner(c *kabanerov1alpha2.Stack, k *kabanerov1alpha2.Kabanero) error {
+	foundKabanero := false
+	ownerReferences := c.GetOwnerReferences()
+	if ownerReferences != nil {
+		for _, ownerRef := range ownerReferences {
+			if ownerRef.Kind == "Kabanero" {
+				if ownerRef.UID == k.ObjectMeta.UID {
+					foundKabanero = true
+				}
+			}
+		}
+	}
+	if !foundKabanero {
+		// Get kabanero instance. Input one does not have APIVersion or Kind.
+		ownerIsController := true
+		kInstance := &kabanerov1alpha2.Kabanero{}
+		name := types.NamespacedName{
+			Name:      k.ObjectMeta.Name,
+			Namespace: c.GetNamespace(),
+		}
+		err := r.client.Get(context.Background(), name, kInstance)
+		if err != nil {
+			return err
+		}
+
+		// Make kabanero the owner of the stack
+		ownerRef := metav1.OwnerReference{
+			APIVersion: kInstance.TypeMeta.APIVersion,
+			Kind:       kInstance.TypeMeta.Kind,
+			Name:       kInstance.ObjectMeta.Name,
+			UID:        kInstance.ObjectMeta.UID,
+			Controller: &ownerIsController,
+		}
+		c.SetOwnerReferences(append(c.GetOwnerReferences(), ownerRef))
+		err = r.client.Update(context.Background(), c)
+		if err != nil {
+			return err
+		}
+		log.Info("Updated stack owner")
+	}
+	return nil
+}
+
+// Used internally by ReconcileStack to store matching stacks
+type resolvedStack struct {
+	repositoryURL string
+	stack    Stack
+}
+
+// ReconcileStack activates or deactivates the input stack.
+func (r *ReconcileStack) ReconcileStack(c *kabanerov1alpha2.Stack, k *kabanerov1alpha2.Kabanero) (reconcile.Result, error) {
+	r_log := log.WithValues("Request.Namespace", c.GetNamespace()).WithValues("Request.Name", c.GetName())
+
+	// Clear the status message, we'll generate a new one if necessary
+	c.Status.StatusMessage = ""
+
+	//The stack name can be either the spec.name or the resource name. The
+	//spec.name has precedence
+	var stackName string
+	if c.Spec.Name != "" {
+		stackName = c.Spec.Name
+	} else {
+		stackName = c.Name
+	}
+
+	r_log = r_log.WithValues("Stack.Name", stackName)
+
+	// A stack created by the CLI might not have kabanero as the owner.
+	// In that case we want to make kabanero the owner
+	err := r.ensureStackHasOwner(c, k)
+	if err != nil {
+		r_log.Error(err, "Could not make kabanero the owner of the stack")
+	}
+
+
+
+	// Retreive all matching stack names from all remote indexes.
+	// TODO: Start using the URL in the Stack object so we don't need to reference the
+	//       parent Kabanero anymore.
+	var matchingStacks []resolvedStack
```

Comment:
ok this whole block here... we don't need to worry about what indexes are in the kabanero CR instance anymore, nor what version they are.  just always iterate the versions in the Stack CR instance, and build out the collection list manually, like you're doing in the `case "v2":` block below.